test/scheduler/tainttoleration_integration_test.go

```diff
@@ -0,0 +1,702 @@
+		})
+
+		AfterAll(func() {
+			// Remove taints
+			removeTaintsFromMemberClusters(taintClusters)
+			// Delete the CRP.
+			ensureCRPAndAllRelatedResourcesDeletion(crpName)
+		})
+	})
+
+	// This is a serial test as adding a new member cluster may interrupt other test cases.
+	Context("pickAll, add a new healthy cluster with taint", Serial, Ordered, func() {
+		crpName := fmt.Sprintf(crpNameTemplate, GinkgoParallelProcess())
+		policySnapshotName := fmt.Sprintf(policySnapshotNameTemplate, crpName, 1)
+		// Prepare a new cluster to avoid interrupting other concurrently running test cases.
+		newUnhealthyMemberClusterName := fmt.Sprintf(provisionalClusterNameTemplate, GinkgoParallelProcess())
+
+		BeforeAll(func() {
+			// Ensure that no bindings have been created so far.
+			noBindingsCreatedActual := noBindingsCreatedForCRPActual(crpName)
+			Consistently(noBindingsCreatedActual, consistentlyDuration, consistentlyInterval).Should(Succeed(), "Some bindings have been created unexpectedly")
+
+			// Create a CRP with no scheduling policy specified, along with its associated policy snapshot.
+			createNilSchedulingPolicyCRPWithPolicySnapshot(crpName, policySnapshotName, nil)
+
+			// Create a new member cluster.
+			createMemberCluster(newUnhealthyMemberClusterName, buildTaints([]string{newUnhealthyMemberClusterName}))
+
+			// Mark this cluster as healthy.
+			markClusterAsHealthy(newUnhealthyMemberClusterName)
+		})
+
+		It("should create scheduled bindings for existing clusters, and exclude new cluster with taint", func() {
+			scheduledBindingsCreatedActual := scheduledBindingsCreatedOrUpdatedForClustersActual(healthyClusters, zeroScoreByCluster, crpName, policySnapshotName)
+			Eventually(scheduledBindingsCreatedActual, eventuallyDuration, eventuallyInterval).Should(Succeed(), "Failed to create the expected set of bindings")
+			Consistently(scheduledBindingsCreatedActual, consistentlyDuration, consistentlyInterval).Should(Succeed(), "Failed to create the expected set of bindings")
+		})
+
+		AfterAll(func() {
+			// Delete the CRP.
+			ensureCRPAndAllRelatedResourcesDeletion(crpName)
+			// Delete the provisional cluster.
+			ensureProvisionalClusterDeletion(newUnhealthyMemberClusterName)
+		})
+	})
+
+	// This is a serial test as adding a new member cluster may interrupt other test cases.
+	Context("pickAll, add a new healthy cluster with taint and matching toleration", Serial, Ordered, func() {
+		crpName := fmt.Sprintf(crpNameTemplate, GinkgoParallelProcess())
+		policySnapshotName := fmt.Sprintf(policySnapshotNameTemplate, crpName, 1)
+		// Prepare a new cluster to avoid interrupting other concurrently running test cases.
+		newUnhealthyMemberClusterName := fmt.Sprintf(provisionalClusterNameTemplate, GinkgoParallelProcess())
+		updatedHealthyClusters := healthyClusters
+		updatedHealthyClusters = append(updatedHealthyClusters, newUnhealthyMemberClusterName)
```

Comment:
did not get it, the newUnhealthyMemberClusterName is unhealthy and why it will be added into the healthy group?

the name is wrong?